public/app/core/components/Tooltip/Popover.tsx

```diff
@@ -1,34 +1,19 @@
-﻿import React from 'react';
-import withTooltip from './withTooltip';
-import { Target } from 'react-popper';
+﻿import React, { PureComponent } from 'react';
+import Popper from './Popper';
+import withPopper, { UsingPopperProps } from './withPopper';
 
-interface PopoverProps {
-  tooltipSetState: (prevState: object) => void;
-}
-
-class Popover extends React.Component<PopoverProps, any> {
-  constructor(props) {
-    super(props);
-    this.toggleTooltip = this.toggleTooltip.bind(this);
-  }
+class Popover extends PureComponent<UsingPopperProps> {
+  render() {
+    const { children, hidePopper, showPopper, className, ...restProps } = this.props;
 
-  toggleTooltip() {
-    const { tooltipSetState } = this.props;
-    tooltipSetState(prevState => {
-      return {
-        ...prevState,
-        show: !prevState.show,
-      };
-    });
-  }
+    const togglePopper = restProps.show === true ? hidePopper : showPopper;
```

Comment:
can be simplified to `restProps.show ? hidePopper : showPopper`

Reply:
Indeed!